### PR TITLE
Delay import of numpy in setup.py

### LINKED ISSUE #147: setup.py install_requires missing numpy
setup.py is missing numpy, installing with pip leads to:
`ImportError: No module named 'numpy'`
running pip with python3.6

### PR DESCRIPTION
**Pull request by estan**
@inducer would something like this be OK? It allows pycuda to be installed before numpy, so it can be pip installed.

Closes #147

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -101,9 +101,33 @@ def get_config_schema():
...
+                if numpy_incpath not in extension.include_dirs:
+                    extension.include_dirs.append(numpy_incpath)
+            super().build_extension(extension)
+
+        def get_numpy_incpath(self):
+            from imp import find_module
+            # avoid actually importing numpy, it screws up distutils
+            file, pathname, descr = find_module("numpy")
+            from os.path import join
+            return join(pathname, "core", "include")
```
Comment (kalvdans) [inline setup.py]: Minor note: `get_numpy_incpath` doesn't access any instance variables and can therefore be defined outside the `custom_build_ext` class.

Comment (estan) [reply setup.py]: Ah true. I could even have reused the `NumpyExtension.get_numpy_incpath` in `aksetup_helper.py` from which it was copied (through would have to make an instance of its `NumpyExtension`). I'll wait and see what @inducer says, he may even want me to patch `aksetup_helper.py` with this stuff instead.

**Comment by inducer** [comment]
Thanks for your PR! I've gone a slightly different route in a6551dd442ed626d394d3431daa2fb1d1d3f3184, but I still appreciate your input--seeing what you did made my life easier!

**Comment by estan** [comment]
@inducer Excellent, many thanks.

**Pull request closed without merging**